windows/RNLocalize/RNLocalizeModule.cpp

```diff
@@ -0,0 +1,190 @@
+#include "pch.h"
+
+#include "RNLocalizeModule.h"
+
+#include "winrt/Windows.Globalization.h"
+
+using namespace winrt::Microsoft::ReactNative;
+
+namespace winrt::RNLocalize
+{
+
+std::vector<winrt::hstring> RNLocalizeModule::getLocaleNames()
+{
+    std::vector<winrt::hstring> locales;
+    for (winrt::hstring locale : winrt::Windows::Globalization::ApplicationLanguages::Languages())
+    {
+        locales.push_back(locale);
+    }
+    return locales;
+}
+
+std::vector<Locale> RNLocalizeModule::getLocales(std::vector<winrt::hstring> localeStrings)
+{
+    std::vector<Locale> localesArray;
+    for (winrt::hstring localeString : localeStrings)
+    {
+        Locale locale;
+        std::string languageCode = getLanguageCode(to_string(localeString));
+        locale.languageCode = languageCode;
+
+        std::string countryCode = getCountryCode(to_string(localeString));
+        locale.countryCode = countryCode;
+
+        locale.isRTL = getIsRTL(languageCode);
+
+        std::string scriptCode = getScriptCode(localeString);
+        locale.scriptCode = scriptCode;
+
+        std::string languageTag = languageCode + "-" + scriptCode + "-" + countryCode;
+        locale.languageTag = languageTag;
+
+        localesArray.push_back(locale);
+    }
+    return localesArray;
+}
+
+std::string RNLocalizeModule::getCountryCode(std::string locale)
+{
+    return locale.substr(locale.find("-") + 1, locale.length());
+}
+
+std::string RNLocalizeModule::getLanguageCode(std::string locale)
+{
+    return locale.substr(0, locale.find("-"));
+}
+
+bool RNLocalizeModule::getIsRTL(std::string languageCode)
+{
+    return std::find(USES_RTL_LAYOUT.begin(), USES_RTL_LAYOUT.end(), languageCode) != USES_RTL_LAYOUT.end();
+}
+
+bool RNLocalizeModule::getUsesMetricSystem(std::string countryCode)
+{
+    return std::find(USES_IMPERIAL.begin(), USES_IMPERIAL.end(), countryCode) == USES_IMPERIAL.end();
+}
+
+std::string RNLocalizeModule::getTemperatureUnit(std::string countryCode)
+{
+    return std::find(USES_FAHRENHEIT.begin(), USES_FAHRENHEIT.end(), countryCode) != USES_FAHRENHEIT.end() ? "fahrenheit" : "celsius";
```

Comment:
y no kelvin! :P

Reply:
I was trying to stay consistent with rest of module and I couldn't find any country that uses kelvin as their standard temperature unit. :P